### ISSUE TITLE
Directives

### ISSUE THREAD

**Issue by eignnx** [eignnx/rellog#15]
Current thinking:

## Syntax
`[[blah][blarg][bleh]]`

## Impl
An `Item::Directive` stores a `Rel`, let's also have it store an `Option<Clause>` which is the next clause that (immediately) follows the directive

## Directive Access
Add some builtin like `[directive][clause]` which can query all directives with a non-`None` following clause. And change `[directive]` (relies on #3) to only match the ones that don't have a following clause.

## Remaining Questions
When is "compile time"? We need a mechanism for running code while a module is being loaded.

Maybe that looks something like:

```
[[loadtime
    - [directive [Meta]][clause [Head][body _]]
    - [define_rel ...][define_body ...]
    - [define_fact ...]
]]

[[module_info M][module_load_hook
    ...
]]
```

**Comment by eignnx** [comment]
`[[loadtime]]` should run the goal it's supplied repeatedly until no choicepoints are left. That way you don't have to `maplist`.

**Comment by eignnx** [comment]
This would also be a good place to add:
- [ ] `[[import]]`
- [ ] `[def_clause [head][body]]` (only accessible from inside `loadtime` contexts for now)
- [ ] `[clause [head][body]]` (only accessible from inside `loadtime` contexts for now)

**Comment by eignnx** [comment]
closed by 1f8fc4b
